src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -283,6 +301,8 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("(?:abc*|def*)g", "(?:ab(?>c*)|de(?>f*))g")]
         [InlineData("(?:a[ce]*|b*)g", "(?:a(?>[ce]*)|(?>b*))g")]
         [InlineData("(?:a[ce]*|b*)c", "(?:a[ce]*|(?>b*))c")]
+        [InlineData("apple|(?:orange|pear)|grape", "apple|orange|pear|grape")]
+        [InlineData("(?>(?>(?>(?:abc)*)))", "(?:abc)*")]
```

Comment:
Not sure whose idea it was but this idea of comparing regex codes is actually super helpful for understanding these tree reduction changes (and protecting them)

Reply:
It would be nice, though, if we could protect the direction of the changes, ie., verifying that "a|b" became "[ab]" and not vice versa (nor both became some 3rd equivalent thing).

Reply:
> Not sure whose idea it was but this idea of comparing regex codes is actually super helpful for understanding these tree reduction changes (and protecting them)

Thanks ;)

> It would be nice, though, if we could protect the direction of the changes

Yes.  For that, we'd really want the ability to output a RegexNode tree back into a regular expression, such that we could then compare the reduced expression to what we expect it to be.